app/models/parent_object.rb

```diff
@@ -13,6 +13,7 @@ class ParentObject < ApplicationRecord # rubocop:disable Metrics/ClassLength
   has_many :batch_processes, through: :batch_connections
   belongs_to :authoritative_metadata_source, class_name: "MetadataSource"
   attr_accessor :metadata_update
+  attr_accessor :from_mets
```

Comment:
I  think it might be valuable to know that a record came from mets permanently (aka a db field) in stead of transiently. If that's the case, then swapping this for a property on the parent object and then changing it so its set on create instead of in the background jobs might make sense?